File: FACULDADE/algoritmos/PYTHON/LISTA02/questao2.py


class Questao2:

    def __init__(self, numero):
        self.__numero = numero

    @staticmethod
    def apresentacao():
        return ("Questão 2:\n\nEfetuar a leitura de um número e apresentá-lo como o seu módulo (somente seu valor"
                "absoluto) elaborando os cálculos matemáticos para isso.\n")

    def __str__(self):
        self.__numero = abs(self.__numero)
        return f"{self.__numero}"


if __name__ == '__main__':

    print(Questao2.apresentacao())
    numero = int(input("Digite um valor: "))
    print(Questao2(numero))

File: ALGORITMOS E PROGRAMAÇÃO/C/Lista 2/questao2.c
#include <stdio.h>
#include <locale.h>
//#include <math.h>


void apresentacao() 
{
	printf("Questão 2\nEfetuar a leitura de um número e apresentá-lo como o seu módulo (somente seu valor"
		"absoluto) elaborando os cálculos matemáticos para isso.\n\n");
}
void validanumero(int status, double* numero) 
{
	int input, temp;
	while (status != 1) {
		while ((temp = getchar()) != EOF && temp != '\n');
		printf("Input invalido. Digite um novo valor: ");
		status = scanf("%lf", numero);
	}
}

void peganumero(double* numero) 
{
	int status;
	printf("Digite o valor de um número: ");
	status = scanf("%lf", numero);
	validanumero(status, numero);
}

void calculaabs(double* numero) 
{
	//*numero = fabs(*numero);
	if (*numero < 0)
	{
		*numero = -(*numero);
	}
}

void imprimeresultado(double numero) 
{
	printf("%.2lf", numero);
}

int main() 
{
	setlocale(LC_ALL, "");

	double numero;

	apresentacao();
	peganumero(&numero);
	calculaabs(&numero);
	imprimeresultado(numero);
}